### PR TITLE
Add support for term meta data. 

### PR DESCRIPTION
**Pull request by stephenharris**
WordPress 4.4 adds term meta data, and [trac ticket 34602](https://core.trac.wordpress.org/ticket/34062), adds term meta data to the WXR export file. Although that ticket is still open, and is unlikely to be in 4.4, this PR adds support for importing term meta data, following the schema set out in that ticket.

### CODE REVIEW

**Comment by anhskohbo** [comment]
+1

**Comment by rmccue** [comment]
Thanks for the PR, and apologies for the delayed response! Going to hold off on this until the core one is merged. :)

**Comment by livinglab** [comment]
https://core.trac.wordpress.org/changeset/37240

**Comment by boonebgorges** [comment]
https://core.trac.wordpress.org/changeset/37240

**Comment by swissspidy** [comment]
Pending a final review this looks good to me.

**Comment by shivapoudel** [comment]
@rmccue Isn't the core one is merged till date :shipit:

**Comment by JJJ** [comment]
This still looks good!

File: class-wxr-importer.php
Comment (rmccue) [inline class-wxr-importer.php]: This should be `if ( ! $key ) { continue' }`

File: class-wxr-importer.php
Comment (rmccue) [inline class-wxr-importer.php]: Is this an existing filter in the v1 importer? If not, should be `wxr_importer.process_term_meta.key`. Needs documentation too.

If it's not an existing filter, not sure what it offers over the above `wxr_importer.pre_process.term_meta` filter.

File: class-wxr-importer.php
Comment (rmccue) [inline class-wxr-importer.php]: Should pass `$term` here too.

File: class-wxr-importer.php
Comment (rmccue) [inline class-wxr-importer.php]: Is this also an existing action?

File: class-wxr-importer.php
Comment (rmccue) [inline class-wxr-importer.php]: `else if` -> `elseif`

**Review by rmccue — CHANGES_REQUESTED** [review]

**Comment by rmccue** [comment]
@stephenharris Sorry for the long delay here :( Submitted a review; if you don't have time to fix this up, I can do so for you.

File: class-wxr-importer.php
Comment (JJJ) [inline class-wxr-importer.php]: Agree.

File: class-wxr-importer.php
Comment (JJJ) [inline class-wxr-importer.php]: Agree.

File: class-wxr-importer.php
Comment (JJJ) [inline class-wxr-importer.php]: Agree.

File: class-wxr-importer.php
Comment (JJJ) [inline class-wxr-importer.php]: Good catch.

File: class-wxr-importer.php
Comment (stephenharris) [inline class-wxr-importer.php]: Happy too, but I wrote this to be consistent with post meta, so we ought to change https://github.com/humanmade/WordPress-Importer/blob/master/class-wxr-importer.php#L1146 too.

(Is backwards compatibility a concern here? If not, we could just replace `$term_id` with `$term`, and similarly for posts).

File: class-wxr-importer.php
Comment (stephenharris) [inline class-wxr-importer.php]: The name is adapted from https://github.com/humanmade/WordPress-Importer/blob/master/class-wxr-importer.php#L1151

In view of that, do we still want the name changed? What about post meta?

File: class-wxr-importer.php
Comment (stephenharris) [inline class-wxr-importer.php]: See earlier comments :)

File: class-wxr-importer.php
Comment (stephenharris) [inline class-wxr-importer.php]: Actually, these hooks are in the v1 importer - but they were added after this PR was opened.

**Comment by stephenharris** [comment]
I clearly messed up re-basing this. I'm going to squash my commits into one and forcibly push it onto my feature branch...

**Comment by stephenharris** [comment]
Decided to open a new PR for this: #106 

Sorry for the noise all.

**Comment by rmccue** [comment]
In future, please don't rebase, just merge `master` into your branch regularly :)

**Pull request closed without merging**
